File: cryptocurrencyWeb/src/main/java/com/outsourcebooster/cryptocurrency/web/config/security/controller/AuthenticationRestController.java
package com.outsourcebooster.cryptocurrency.web.config.security.controller;


import com.outsourcebooster.cryptocurrency.web.config.security.model.JwtAuthenticationRequest;
import com.outsourcebooster.cryptocurrency.web.config.security.util.JwtTokenUtil;
import com.outsourcebooster.cryptocurrency.web.config.security.model.JwtAuthenticationResponse;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.http.ResponseEntity;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.AuthenticationException;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class AuthenticationRestController {

    @Value("${jwt.header}")
    private String tokenHeader;

    @Autowired
    private AuthenticationManager authenticationManager;

    @Autowired
    private JwtTokenUtil jwtTokenUtil;

    @Autowired
    private UserDetailsService userDetailsService;

    @RequestMapping(value = "auth", method = RequestMethod.POST)
    public ResponseEntity<JwtAuthenticationResponse> createAuthenticationToken(@RequestBody JwtAuthenticationRequest authenticationRequest) throws AuthenticationException {
        UsernamePasswordAuthenticationToken usernamePasswordAuthenticationToken = new UsernamePasswordAuthenticationToken(
                authenticationRequest.getEmail(),
                authenticationRequest.getPassword()
        );
        Authentication authentication = authenticationManager.authenticate(usernamePasswordAuthenticationToken);
        SecurityContextHolder.getContext().setAuthentication(authentication);

        UserDetails userDetails = userDetailsService.loadUserByUsername(authenticationRequest.getEmail());
        String token = jwtTokenUtil.generateToken(userDetails);

        return ResponseEntity.ok(new JwtAuthenticationResponse(token));
    }
}

File: cryptocurrencyWeb/src/main/java/com/outsourcebooster/cryptocurrency/web/config/security/controller/AuthenticationRestController.kt
package com.outsourcebooster.cryptocurrency.web.config.security.controller;


import com.outsourcebooster.cryptocurrency.web.config.security.model.JwtAuthenticationRequest;
import com.outsourcebooster.cryptocurrency.web.config.security.util.JwtTokenUtil;
import com.outsourcebooster.cryptocurrency.web.config.security.model.JwtAuthenticationResponse;
import com.outsourcebooster.cryptocurrency.web.service.user.UserService
import org.springframework.http.ResponseEntity;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

@RestController
open class AuthenticationRestController(private val authenticationManager: AuthenticationManager,
                                        private val jwtTokenUtil: JwtTokenUtil,
                                        private val userDetailsService: UserDetailsService,
                                        private val userService: UserService) {


    @RequestMapping(value = "auth", method = arrayOf(RequestMethod.POST))
    fun  createAuthenticationToken(@RequestBody  authenticationRequest: JwtAuthenticationRequest): ResponseEntity<JwtAuthenticationResponse> {
        val usernamePasswordAuthenticationToken = UsernamePasswordAuthenticationToken(
                authenticationRequest.email,
                authenticationRequest.password
        )
        val authentication = authenticationManager.authenticate(usernamePasswordAuthenticationToken)
        SecurityContextHolder.getContext().setAuthentication(authentication)
        val userDetails = userDetailsService.loadUserByUsername(authenticationRequest.email)
        val token = jwtTokenUtil.generateToken(userDetails)
        val user = userService.getUserByEmail(authenticationRequest.email)
        user!!.password = ""
        return ResponseEntity.ok(JwtAuthenticationResponse(token, user))
    }
}
